src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java

```diff
@@ -125,17 +125,21 @@ public void move(@NonNegative long from, @NonNegative long to, @NonNegative long
 
     @NotNull
     @Override
-    public Bytes<U> compact()
-            throws IllegalStateException {
+    public Bytes<U> compact() {
         assert DISABLE_SINGLE_THREADED_CHECK || threadSafetyCheck(true);
-        long start = start();
-        long readRemaining = readRemaining();
-        if ((readRemaining > 0) && (start < readPosition)) {
-            bytesStore.move(readPosition, start, readRemaining);
+        try {
+            long start = start();
+            long readRemaining = readRemaining();
+            if ((readRemaining > 0) && (start < readPosition)) {
+                bytesStore.move(readPosition, start, readRemaining);
+            }
             readPosition = start;
-            uncheckedWritePosition(readPosition + readRemaining);
+            uncheckedWritePosition(start + Math.max(0, readRemaining));
+
+            return this;
+        } catch (IllegalStateException ignored) {
```

Comment:
should be ClosedIllegalStateException? Should we have a test for calling compact on a released Bytes?

Reply:
compact() should fail fast.
While it could ignore being called after closing, this is almost certainly a coding error.

Reply:
Don't understand.... If we want to fail fast, why are we catching and ignoring the ISE?